ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -152,6 +166,43 @@ public StoreInfo get(List<? extends StoreKey> ids, EnumSet<StoreGetOptions> stor
     }
   }
 
+  /**
+   * Checks the state of the messages in the given {@link MessageWriteSet} in the given {@link FileSpan}.
+   * @param messageSetToWrite Non-empty set of messages to write to the store.
+   * @param fileSpan The fileSpan on which the check for existence of the messages have to be made.
+   * @return {@link MessageWriteSetStateInStore} representing the outcome of the state check.
+   * @throws StoreException relays those encountered from {@link PersistentIndex#findKey(StoreKey, FileSpan)}.
```

Comment:
Removed the check that throws this exception, since these are private methods and the checks are the caller's responsibility.